ClemBot.Bot/bot/cogs/trivia_cog.py

```diff
@@ -0,0 +1,452 @@
+            category_embed.add_field(name="Index:", value=x)
+
+            final_page.append(category_embed)
+
+        difficulty_generator = helper_fixer(DIFFICULTY)
+
+      
+        for y in difficulty_generator:
+            difficulty_embed = discord.Embed(title="Difficulty List:", color=Colors.ClemsonOrange)
+            difficulty_embed.add_field(name="Index:", value=y)
+
+            final_page.append(difficulty_embed)
+            
+
+        question_generator = helper_fixer(QUESTIONTYPE)
+        
+        for z in question_generator:
+            type_embed = discord.Embed(title="Question Type:", color=Colors.ClemsonOrange)
+            type_embed.add_field(name="Index:", value=z)
+
+            final_page.append(type_embed)
+          
+
+        await self.bot.messenger.publish(Events.on_set_pageable_embed,
+                                         pages=final_page,
+                                         author=ctx.author,
+                                         channel=ctx.channel,
+                                         timeout=60,)
+
+    def html_parser(self, new_response):
+
+        dictionary_list = []  #pain
+
+        for x in new_response["results"]:
+            new_dictionary = x
+            new_list_values = x.values()
+            proper_values = []
+
+            for b in new_list_values:
+                if isinstance(b, list):
+                    new_list = []
+                    for y in b:
+                        if not y.isnumeric():
+                            new_list.append(html.unescape(y))  #This HTML response is in a weird format where you have to navigate a list that contains dictionaries with that dictionary containing a SINGLE list
+                        else:
+                            new_list.append(y)
+                    proper_values.append(new_list)
+                elif not b.isnumeric():
+                    proper_values.append(html.unescape(b))  # good luck maintaining this
+                else:
+                    proper_values.append(b)
+
+            proper_values_size = len(proper_values)
+            biggest_loop = 0
+
+            for key, value in new_dictionary.items():  #.items is special because it contains an active view
+                new_dictionary[key] = proper_values[biggest_loop]
+                if biggest_loop < proper_values_size:
+                    biggest_loop += 1
+                else:
+                    break
+
+            dictionary_list.append(new_dictionary)
+
+        dictionary_size = len(new_response["results"])
+       
+        
+        for best_loopint in range (0,dictionary_size):
+            new_response["results"][best_loopint] = dictionary_list[best_loopint]  #Sets the real dictionary to our parsed results
+
+        return new_response
+
+    async def dict_publisher(self, dictionary):
+
+        x = 0
+        cog_embeds = []
+        list_index = []
+
+        for create_lists in dictionary["results"]:
```

Comment:
Here instead of manually incrementing `x`, you should use Python's `enumerate` function. https://realpython.com/python-enumerate/